### ISSUE TITLE
Python bindings incompatible with sip 4.19+

### ISSUE THREAD

**Issue by tkemmer** [BALL-Project/ball#606]
Since sip 4.19, the `BALLCoremodule` target fails with
```
sip: File::CannotWrite has not been defined
make[3]: *** [source/PYTHON/EXTENSIONS/CMakeFiles/BALLCoremodule.dir/build.make:310: source/PYTHON/EXTENSIONS/BALLCoremodule/sipBALLCorepart0.cpp] Error 1
```

After removing all `throw(File::CannotWrite)` declarations from the sip files (or replacing them by `throw()`, the code can be successfully generated. However, compilation still fails with thousands of error messages in the form of
```
build/source/PYTHON/EXTENSIONS/BALLCoremodule/sipAPIBALLCore.h:4977:39: error: ‘sipExportedExceptions_BALLCore’ was not declared in this scope                                                                                                                                                        
     #define sipException_GeneralException sipExportedExceptions_BALLCore[0]

source/PYTHON/EXTENSIONS/BALL/exception.sip:19:21: note: in expansion of macro ‘sipException_GeneralException’
     PyErr_SetString(sipException_GeneralException, detail.c_str())
```
for all BALL exceptions.

Possible hints:
http://pyqt.sourceforge.net/Docs/sip4/incompatibilities.html#sip-v4-19

**Comment by mitya57** [comment]
If you seek help with porting to new SIP, I suggest you to ask for it on the [PyQt mailing list][1]. Phil and other people there are usually very responsive.

Because of this issue I cannot proceed with transition to new SIP in Ubuntu 17.10.

[1]: https://www.riverbankcomputing.com/mailman/listinfo/pyqt

**Comment by azerbe** [comment]
Same error for the VIEWModule
`
sip: File::CannotWrite has not been defined
make[2]: *** [source/PYTHON/EXTENSIONS/CMakeFiles/VIEWmodule.dir/build.make:310: source/PYTHON/EXTENSIONS/VIEWmodule/sipVIEWpart0.cpp] Fehler 1
make[2]: *** Datei „source/PYTHON/EXTENSIONS/VIEWmodule/sipVIEWpart0.cpp“ wird gelöscht
make[1]: *** [CMakeFiles/Makefile2:7153: source/PYTHON/EXTENSIONS/CMakeFiles/VIEWmodule.dir/all] Fehler 2
`

**Comment by tkemmer** [comment]
The second set of errors (`‘sipExportedExceptions_BALLCore’ was not declared in this scope`) is actually an upstream bug in sip versions 4.19.{0,1,2} and has been fixed in 4.19.3.
